packages/jaeger-ui/src/components/TracePage/TraceSpanView/index.tsx

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { Component } from 'react';
+import { Row, Col, Table, Button, Select } from 'antd';
+import moment from 'moment';
+import { ColumnProps } from 'antd/es/table';
+import { SelectValue } from 'antd/lib/select';
+import FormItem from 'antd/lib/form/FormItem';
+import './index.css';
+import { TNil } from '../../../types';
+import { Trace, Span } from '../../../types/trace';
+
+const Option = Select.Option;
+
+function getNestedProperty(path: string, span: any): string {
+  return path.split('.').reduce((prev, curr) => {
+    return prev ? prev[curr] : null;
+  }, span);
+}
+
+function timeConversion(microseconds: number) {
```

Comment:
I think we can move this to utils (outside the component) and write a couple of unit tests.

Reply:
Ok will take this